### PR TITLE
Auth (close #3, users)

### LINKED ISSUE #3: 3-add-users
```
As a person,
So that I can use Apptivist,
I would like to create a profile.
```

### PR DESCRIPTION
**Pull request by robota-x**
users can log in thorugh github integration

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -57,4 +60,26 @@ app.use(function(err, req, res, next) {
   });
 });
 
+passport.use(new GithubStrategy({
+    clientID: "e2ffec479fa621ef20b4",
+    clientSecret: process.env.GIT_SECRET,
+    callbackURL: "http://localhost:3000/users/auth/callback"
+  },
+  function(accessToken, refreshToken, profile, done) {
+    console.log("verifyF with", profile.username);
```
Comment (robota-x) [inline app.js]: debugging code

File: app.js [hunk 2]
```diff
@@ -57,4 +60,26 @@ app.use(function(err, req, res, next) {
...
+  }
+));
+
+passport.serializeUser(function(user, done) {
+  var token = user.id;
+  done(null, token);
+});
+
+passport.deserializeUser(function(user, done) {
+  // console.log("deserializeUser with", user);
```
Comment (robota-x) [inline app.js]: comments

File: npm-debug.log [hunk 3]
```diff
@@ -0,0 +1,45 @@
+0 info it worked if it ends with ok
```
Comment (robota-x) [inline npm-debug.log]: add this file to ignore

File: server/routes/events.js [hunk 4]
```diff
@@ -19,12 +19,13 @@ router.post('/', function(req, res) {
 });
 
 router.get('/:event_id', function(req, res) {
+  console.log('passing id',req.user);
```
Comment (robota-x) [inline server/routes/events.js]: debugging code

File: server/routes/index.js [hunk 5]
```diff
@@ -3,6 +3,7 @@ var router = express.Router();
 var app = express();
 
 router.get('/', function (req, res) {
+  console.log('loading index, passing i:',req.user);
```
Comment (robota-x) [inline server/routes/index.js]: debugging code

File: server/routes/users.js [hunk 6]
```diff
@@ -1,8 +1,44 @@
...
 
 router.get('/', function(req, res, next) {
   res.send('respond with a resource');
 });
 
+router.get('/auth', passport.authenticate('github'));
+
+router.get('/auth/callback', passport.authenticate('github', { failureRedirect: '/' }),
+  function(req, res) {
+    console.log("success for", req.user.username);
```
Comment (robota-x) [inline server/routes/users.js]: debugging code

File: server/routes/users.js [hunk 7]
```diff
@@ -1,8 +1,44 @@
...
+        } else {
+          res.redirect('/');
+        }
+
+      });
+  }
+);
+
+router.get('/logout', function(req, res){
+  console.log('logging out');
```
Comment (lorenzoturrino) [inline server/routes/users.js]: debugging code

**Pull request merged**